src/protocol/primitives/record-batch/record-batch.spec.ts

```diff
@@ -239,9 +247,55 @@ describe('RecordBatch', () => {
     const deserialized = await RecordBatch.deserialize(new ReadBuffer(buffer));
 
     expect(deserialized).toBeDefined();
-    expect(deserialized.records.value).toHaveLength(1);
-    expect(deserialized.records.value![0]!.key.value).toEqual(Buffer.from([0x01, 0x02, 0x03, 0x04, 0x05, 0x06]));
-    expect(deserialized.records.value![0]!.value.value).toEqual(Buffer.from([0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c]));
+    expect(deserialized.records).toHaveLength(1);
+    expect(deserialized.records![0]!.key.value).toEqual(Buffer.from([0x01, 0x02, 0x03, 0x04, 0x05, 0x06]));
+    expect(deserialized.records![0]!.value.value).toEqual(Buffer.from([0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c]));
     expect(deserialized.attributes.compressionType).toEqual(compressionType);
   });
+
+  it('should throw an error if records array is empty', () => {
+    expect(() =>
+      RecordBatch.from({
+        baseOffset: new Int64(Int64.MIN_VALUE),
+        partitionLeaderEpoch: new Int32(Int32.MIN_VALUE),
+        attributes: {
+          compressionType: CompressionType.ZSTD,
+          timestampType: TimestampType.CreateTime,
+          isTransactional: IsTransactional.Yes,
+          isControlBatch: IsControlBatch.No,
+          hasDeleteHorizon: HasDeleteHorizon.Yes
+        },
+        lastOffsetDelta: new Int32(Int32.MIN_VALUE),
+        baseTimestamp: new Int64(Int64.MIN_VALUE),
+        maxTimestamp: new Int64(Int64.MIN_VALUE),
+        producerId: new Int64(Int64.MIN_VALUE),
+        producerEpoch: new Int16(Int16.MIN_VALUE),
+        baseSequence: new Int32(Int32.MIN_VALUE),
+        records: []
+      })
+    ).toThrow();
+  });
+
+  it('should throw if deserialized length of records is smaller than 0', async () => {
+    const buffer = Buffer.from(
+      '80000000000000000000003a8000000002ce38005a005480000000800000000000000080000000000000008000000000000000800080000000ffffffff28b52ffd2000010000',
```

Comment:
Is there a reason for introducing the new style for providing the buffers?

Reply:
Laziness